types/filterfunctions.html

```diff
@@ -0,0 +1,149 @@
+        margin-bottom: 20px;
+      }
+
+      input {
+        width: 60%;
+      }
+
+      output {
+        width: 5%;
+        text-align: center;
+      }
+
+      select {
+        width: 40%;
+        margin-left: 2px;
+      }
+    </style>
+  </head>
+  <body>
+    <div></div>
+    <ul>
+      <li>
+        <label for="filter-select">Choose a filter function:</label>
+        <select id="filter-select">
+          <option selected>blur</option>
+          <option>brightness</option>
+          <option>contrast</option>
+          <option>drop-shadow</option>
+          <option>grayscale</option>
+          <option>hue-rotate</option>
+          <option>invert</option>
+          <option>opacity</option>
+          <option>saturate</option>
+          <option>sepia</option>
+        </select>
+      </li>
+      <li><input type="range" /><output></output></li>
+      <li>
+        <p>Current value: <code></code></p>
+      </li>
+    </ul>
+    <script>
+      const selectElem = document.querySelector("select");
+      const divElem = document.querySelector("div");
+      const slider = document.querySelector("input");
+      const output = document.querySelector("output");
+      const curValue = document.querySelector("p code");
+
+      selectElem.addEventListener("change", () => {
+        setSlider(selectElem.value);
+        setDiv(selectElem.value);
+      });
+
+      slider.addEventListener("input", () => {
+        setDiv(selectElem.value);
+      });
+
+      function setSlider(filter) {
+        if (filter === "blur") {
+          slider.value = 0;
+          slider.min = 0;
+          slider.max = 30;
+          slider.step = 1;
+          slider.setAttribute("data-unit", "px");
+        } else if (
+          filter === "brightness" ||
+          filter === "contrast" ||
+          filter === "saturate"
+        ) {
+          slider.value = 1;
+          slider.min = 0;
+          slider.max = 4;
+          slider.step = 0.05;
+          slider.setAttribute("data-unit", "");
+        } else if (filter === "drop-shadow") {
+          slider.value = 0;
+          slider.min = -20;
+          slider.max = 40;
+          slider.step = 1;
+          slider.setAttribute("data-unit", "px");
+        } else if (filter === "opacity") {
+          slider.value = 1;
+          slider.min = 0;
+          slider.max = 1;
+          slider.step = 0.01;
+          slider.setAttribute("data-unit", "");
+        } else if (
+          filter === "grayscale" ||
+          filter === "invert" ||
+          filter === "sepia"
+        ) {
+          slider.value = 0;
+          slider.min = 0;
+          slider.max = 1;
+          slider.step = 0.01;
+          slider.setAttribute("data-unit", "");
+        } else if (filter === "hue-rotate") {
+          slider.value = 0;
+          slider.min = 0;
+          slider.max = 360;
+          slider.step = 1;
+          slider.setAttribute("data-unit", "deg");
+        }
+      }
```

Comment:
I think this is easier to read with a `switch` statement:
```suggestion
      function setSlider(filter) {
        switch(filter) {
          case "blur":
            slider.value = 0;
            slider.min = 0;
            slider.max = 30;
            slider.step = 1;
            slider.setAttribute("data-unit", "px");
            break;
          case "brightness":
          case "contrast":
          case"saturate":
            slider.value = 1;
            slider.min = 0;
            slider.max = 4;
            slider.step = 0.05;
            slider.setAttribute("data-unit", "");
            break;
          case "drop-shadow":
            slider.value = 0;
            slider.min = -20;
            slider.max = 40;
            slider.step = 1;
            slider.setAttribute("data-unit", "px");
            break;
          case "opacity":
            slider.value = 1;
            slider.min = 0;
            slider.max = 1;
            slider.step = 0.01;
            slider.setAttribute("data-unit", "");
            break;
          case "grayscale":
          case "invert":
          case "sepia":
            slider.value = 0;
            slider.min = 0;
            slider.max = 1;
            slider.step = 0.01;
            slider.setAttribute("data-unit", "");
            break;
          case "hue-rotate":
            slider.value = 0;
            slider.min = 0;
            slider.max = 360;
            slider.step = 1;
            slider.setAttribute("data-unit", "deg");
            break;
          default:
            console.error("Unknown filter set");
      }
```

I also added a default entry, just in case.

Reply:
(Note that we could also use an array with all the values and read the value from the array. That way, updating the array would be the only thing to do to add more entries or to make entries like `"grayscale"` and `"invert"` perform something different. The code would be shorter too).

This is likely overkill, though.

Reply:
This is a copy / paste from the MDN directory. I wasn't wanting to spend time  on it; i just wanted to port it over.